src/IterableObject.php

```diff
@@ -18,113 +19,50 @@ final class IterableObject implements IteratorAggregate
     /**
      * @var callable
      */
-    private $filter;
+    private $filterFn;
 
     /**
      * @var callable
      */
-    private $map;
+    private $mapFn;
 
-    /**
-     * IterableObject constructor.
-     * @param iterable|array|Traversable $iterable
-     * @param callable|null              $filter
-     * @param callable|null              $map
-     * @throws InvalidArgumentException
-     */
-    public function __construct($iterable, $filter = null, $map = null)
+    public function __construct(?iterable $iterable = null, ?callable $filter = null, ?callable $map = null)
     {
-        if (null === $iterable) {
-            $iterable = new EmptyIterator();
-        }
-        if (!is_iterable($iterable)) {
-            throw new InvalidArgumentException(
-                sprintf('Expected array or Traversable, got %s', is_object($iterable) ? get_class($iterable) : gettype($iterable))
-            );
-        }
-
-        // Cannot rely on callable type-hint on PHP 5.3
-        if (null !== $filter && !is_callable($filter) && !$filter instanceof Closure) {
-            throw new InvalidArgumentException(
-                sprintf('Expected callable, got %s', is_object($filter) ? get_class($filter) : gettype($filter))
-            );
-        }
-
-        if (null !== $map && !is_callable($map) && !$map instanceof Closure) {
-            throw new InvalidArgumentException(
-                sprintf('Expected callable, got %s', is_object($map) ? get_class($map) : gettype($map))
-            );
-        }
-
-        $this->iterable = $iterable;
-        $this->filter = $filter;
-        $this->map = $map;
+        $this->iterable = $iterable ?? new EmptyIterator();
+        $this->filterFn = $filter;
+        $this->mapFn = $map;
```

Comment:
the names differ here

Reply:
Yes, I renamed internal props, as IDE complained properties had similar names than methods.